documentation/docs/specs/revocation_bitmap_2022.md

```diff
@@ -0,0 +1,196 @@
+
+1. Let **service endpoint** be a data url generated using the [service endpoint generation algorithm](#service-endpoint-generation-algorithm).
+2. The `<mediatype>` of the **service endpoint** MUST be `application/octet-stream` and the `base64` attribute MUST be set, return an error otherwise. Let **compressed bitmap** be the `<data>` part of the data url.
+3. Generate an **uncompressed bitmap** by base64-decoding [[RFC4648](https://datatracker.ietf.org/doc/html/rfc4648)] the **compressed bitmap** and then decompressing the result using ZLIB [[RFC 1950](https://datatracker.ietf.org/doc/html/rfc1950)].
+4. Generate the **bitmap** by deserializing the **uncompressed bitmap** according to the [roaring bitmap serialization format](https://github.com/RoaringBitmap/RoaringFormatSpec/).
+5. Return the **bitmap**.
+
+### Validation Algorithm
+
+The following steps MUST be followed when checking whether a verifiable credential is revoked:
+
+1. Let **credential** be a verifiable credential containing a `credentialStatus` whose `type` is `RevocationBitmap2022`.
+2. Let **revocation bitmap URL** be the `id` field of the **credential**'s `credentialStatus`.
+3. Resolve the **revocation bitmap URL** to a **revocation bitmap service** in the issuer's DID document, and verify that the service `type` is `RevocationBitmap2022`. Return an error otherwise.
+4. Expand the endpoint of the **revocation bitmap service** into a **revocation bitmap** according to the [service endpoint expansion algorithm](#service-endpoint-expansion-algorithm).
+5. Let **revocation index** be the integer value of the `revocationBitmapIndex` property of the **credential**.
+6. Let **revoked** be the value of the bit at index **revocation index** in the **revocation bitmap**.
+7. Return `true` if **revoked** is 1, `false` otherwise.
+
+## Test Vectors
+
+This section provides test vectors to validate implementations against.
+
+### Test Vector 1
+
+The following data URL decodes to a bitmap of length 0 where no index is revoked:
+
+`"data:application/octet-stream;base64,ZUp5ek1tQUFBd0FES0FCcg=="`
+
+### Test Vector 2
+
+The following data URL decodes to a bitmap of length 3 where indices `5`, `398`, and `67000` are revoked:
+
+`"data:application/octet-stream;base64,ZUp5ek1tQmdZR0lBQVVZZ1pHQ1FBR0laSUdabDZHUGN3UW9BRXVvQjlB"`.
+
+### Test Vector 3
+
+The following data URL decodes to a bitmap of length 16384 where all indices are revoked:
+
+`"data:application/octet-stream;base64,ZUp6dHhERVJBQ0FNQkxESEFWS1lXZkN2Q3E0MmFESmtyMlNrM0ROckFLQ2RBQUFBQUFBQTMzbGhHZm9q"`
+
+## Rationale
+
+This section describes the rationale behind some of the design decisions of this specification.
+
+### Compression
+
+Considering that messages published to the Tangle cannot exceed [32 KiB](https://github.com/iotaledger/tips/blob/main/tips/TIP-0006/tip-0006.md#message-validation) in size, and that larger messages have increased requirements, the use of compression was assessed.
+
```

Comment:
Maybe we can mention a ballpark figure of how many revocations are possible to fit on the Tangle with compression here?

```suggestion
Considering that messages published to the Tangle cannot exceed [32 KiB](https://github.com/iotaledger/tips/blob/main/tips/TIP-0006/tip-0006.md#message-validation) in size, and that larger messages have increased requirements, the use of compression was assessed.

The precise size of a serialized bitmap varies based on the number and distribution of revoked indices. With compression, it is possible to achieve on the order of 100,000 - 200,000 in a 32KiB DID Document when indices are revoked uniformly randomly, and significantly more if consecutive indices are revoked.
```